classes/core/PKPPageRouter.php

```diff
@@ -399,81 +400,45 @@ public function redirectHome(PKPRequest $request): void
-            }
-
-            if (Config::getVar('features', 'enable_new_submission_listing')) {
-
-                $roleIdsArray = $userGroups->pluck('role_id')->all();
+        if (Config::getVar('features', 'enable_new_submission_listing')) {
+            $roleIdsArray = $userGroups->pluck('role_id')->all();
 
-                if (array_intersect([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT], $roleIdsArray)) {
-                    return $request->url(null, 'dashboard', 'editorial');
+            if (array_intersect([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT], $roleIdsArray)) {
+                return $request->url(null, 'dashboard', 'editorial');
+            }
 
-                }
-                if (in_array(Role::ROLE_ID_REVIEWER, $roleIdsArray)) {
-                    return $request->url(null, 'dashboard', 'reviewAssignments');
-                }
-                if (in_array(Role::ROLE_ID_AUTHOR, $roleIdsArray)) {
-                    return $request->url(null, 'dashboard', 'mySubmissions');
-                }
+            if (in_array(Role::ROLE_ID_REVIEWER, $roleIdsArray)) {
+                return $request->url(null, 'dashboard', 'reviewAssignments');
             }
 
-            return $request->url(null, 'submissions');
-        } else {
-            // The user is at the site context
-            $userGroups = UserGroup::query()
-                ->where('context_id', \PKP\core\PKPApplication::SITE_CONTEXT_ID)
-                ->whereHas('userUserGroups', function ($query) use ($userId) {
-                    $query->where('user_id', $userId)
-                        ->where(function ($q) {
-                            $q->whereNull('date_end')
-                                ->orWhere('date_end', '>', now());
-                        })
-                        ->where(function ($q) {
-                            $q->whereNull('date_start')
-                                ->orWhere('date_start', '<=', now());
-                        });
-                })
-                ->get();
-
-            if ($userGroups->count() == 1) {
-                $firstUserGroup = $userGroups->first();
-                $contextDao = Application::getContextDAO();
-                $context = $contextDao->getById($firstUserGroup->contextId);
-                if (!isset($context)) {
-                    $request->redirect(Application::SITE_CONTEXT_PATH, 'index');
-                }
-                if ($firstUserGroup->roleId == Role::ROLE_ID_READER) {
-                    $request->redirect(null, 'index');
-                }
+            if (in_array(Role::ROLE_ID_AUTHOR, $roleIdsArray)) {
+                return $request->url(null, 'dashboard', 'mySubmissions');
             }
-            return $request->url(Application::SITE_CONTEXT_PATH, 'index');
         }
-    }
 
+        return $request->url(null, 'submissions');
+    }
```

Comment:
@jonasraoni When this would be hit? /submissions is not used anymore.. just the /dashboard/{mySubmissions|editorial|reviewAssignments}

Reply:
- The user must have:
  A. More than one role on the context
  B. Just one role, but it shouldn't be the reader role
- `[features].enable_new_submission_listing` should be disabled

Reply:
@jonasraoni I see, ok. Thank you